Sources/SwiftBSON/BSONDocumentIterator.swift

```diff
@@ -49,12 +47,13 @@ public class BSONDocumentIterator: IteratorProtocol {
     /// Get the next key in the iterator, if there is one.
     /// This method should only be used for iterating through the keys. It advances to the beginning of the next
     /// element, meaning the element associated with the last returned key cannot be accessed via this iterator.
-    private func nextKey() throws -> String? {
-        guard let type = try self.readNextType() else {
+    private func nextKey() -> String? {
```

Comment:
any methods not called from `nextThrowing` (which is used in validation and thus has useful error messages) I removed `throws` from and instead just returned `nil`.

Reply:
I think it would be helpful for future us to note throughout the docstrings in the file what the method's behaviors are when invalid BSON is encountered

Reply:
done